### DISCUSSION TITLE
CMake Error when running case.build after porting

### DISCUSSION THREAD

**Discussion by chprsandeep** [E3SM-Project/E3SM#4931] [E3SM model help]
I am new to E3SM/ELM. But I have run several cases of CESM2/CLM.

I have ported E3SM to my system and got an error while running the case.build

Here is a step-by-step process of what I have done:

git clone [git@github.com](mailto:git@github.com):E3SM-Project/E3SM.git
cd E3SM
git submodule update --init --recursive

I have copied the configure files from CESM to cime_config/machines/ folder and updated the corresponding directories.

cd cime/scripts
./create_newcase --case /net/fs05/d1/sandeepc/e3sm/cases/ast --compset ICRUELMBGC --res hcru_hcru --machine svante --compiler intel
cd /net/fs05/d1/sandeepc/e3sm/cases/ast

./case.setup
./case.build

I got the following error.
ERROR: BUILD FAIL: cmake config e3sm failed, cat /net/fs05/d1/sandeepc/e3sm/cases/ast/bld/e3sm.bldlog.220429-074949

I have attached the e3sm.bldlog file and configure files.
[e3sm.bldlog.220429-074949.txt](https://github.com/E3SM-Project/E3SM/files/8611589/e3sm.bldlog.220429-074949.txt)
[config_batch.xml.txt](https://github.com/E3SM-Project/E3SM/files/8611579/config_batch.xml.txt)
[config_compilers.xml.txt](https://github.com/E3SM-Project/E3SM/files/8611580/config_compilers.xml.txt)
[config_machines.xml.txt](https://github.com/E3SM-Project/E3SM/files/8611581/config_machines.xml.txt)

Any help would be much appreciated. Please let me know if anything else is required from my side.

Thank you.
Sandeep Chinta

**Comment by rljacob** [comment]
E3SM is no longer using config_compilers.xml.   You need to create an intel_svante.cmake file and put it in E3SM/cime_config/machines/cmake_macros

**Reply by chprsandeep** [reply]
Thank you for the reply.  I will create a .cmake file.

Could you point me to a resource like a user guide or tutorial where I will be able to know about these things? I was not able to find one.

Regards
Sandeep

**Reply by daofeng2007** [reply]
I am trying to port the machine to a Ubuntu GPU server. I also encountered an error in the ./case.build step. I was not able to find E3SM/cime_config/machines/cmake_macros either. @rljacob , could you guide us how to port the machine using cmake_marcros?

**Reply by rljacob** [reply]
Are you using the latest version?  We did not move to cmake until 2.0

**Reply by daofeng2007** [reply]
Thanks for your reply, @rljacob Robert. I used the `git clone -b maint-1.0 --recursive https://github.com/E3SM-Project/E3SM.git` to install the E3SM. It is not the latest version.
I am using[ instructions here](https://esmci.github.io/cime/versions/master/html/users_guide/porting-cime.html) to port the machine to my Ubuntu server. That tutorial indeed mentions cmake. Is this the correct place I should study to port the machine to my platform?

**Reply by rljacob** [reply]
Try these older docs:  https://esmci.github.io/cime/versions/maint-5.6/html/index.html

**Reply by chprsandeep** [reply]
Hi @daofeng2007 

I was able to create a working cmake file for my case using other examples in the folder. You may edit accordingly for your system. Hope this helps.

The following is the content in my intel_svante.cmake file in /home/E3SM/cime_config/machines/cmake_macros directory.

string(APPEND CPPDEFS " -DNO_SHR_VMATH -DCNL")
if (DEBUG)
  string(APPEND FFLAGS " -check all -ftrapuv")
endif()
string(APPEND SLIBS " -llapack -lblas")
set(PYTHONPATH "/home/software/python/3.9.1/")
set(MPI_LIB_NAME "mpi")
set(MPI_PATH "/home/software/intel/intel-2017.0.1/pkg/openmpi/2.0.2/")
set(NETCDF_PATH "/home/software/intel/intel-2017.0.1/pkg/netcdf/4/")
execute_process(COMMAND /home/software/intel/intel-2017.0.1/pkg/netcdf/4/bin/nf-config --flibs OUTPUT_VARIABLE SHELL_CMD_OUTPUT_BUILD_INTERNAL_IGNORE0 OUTPUT_STRIP_TRAILING_WHITESPACE)
string(APPEND SLIBS " ${SHELL_CMD_OUTPUT_BUILD_INTERNAL_IGNORE0}")

**Reply by LianGong98** [reply]
@rljacob Hi , I am also trying  to port E3SM-2.0.0 to my  Ubuntu server.The E3SM is no longer using config_compilers.xml, so what the intel_svante.cmake file should contain? Or is there a template?
Any help would be greatly appreciated!

**Reply by rljacob** [reply]
gnu_anlgce.cmake is an ubuntu machine.  Just copy that to start.
